README.md

```diff
@@ -100,3 +98,64 @@ pyinfra inventory.py deploy.py
 Now you know the building blocks of pyinfra! By combining inventory, operations and Python code you can deploy anything.
 
 See the more detailed [getting started](https://docs.pyinfra.com/page/getting-started.html) or [using operations](https://docs.pyinfra.com/page/using-operations.html) guides. See how to use [inventory & data](https://docs.pyinfra.com/page/inventory-data.html), [global arguments](https://docs.pyinfra.com/page/arguments.html) and [the CLI](https://docs.pyinfra.com/page/cli.html) or check out the [documented examples](https://docs.pyinfra.com/page/examples.html).
+
+## Using Pyinfra Programtically
+
+Pyinfra can also be used directly from within a Python a application, without the need of the CLI.
+Quick example shown below:
+
+```python
+# Basic imports
+from pyinfra.api import Config, Inventory, State
+from pyinfra.api.operation import add_op
+from pyinfra.api.operations import run_ops
+from pyinfra.api.connect import connect_all
+from pyinfra.api.facts import get_facts
+
+
+# Build your inventory:
+inventory = Inventory(
+    (["myhostname"], {
+            "ssh_hostname": "localhost",
+            "ssh_user": "vagrant",
+            "ssh_port": "2222",
+            "ssh_key": "/path/to/ssh_key/file"
+        })
+)
+
+# Initialize the configuration
+config = Config(
+    FAIL_PERCENT=81,
+    CONNECT_TIMEOUT=5
+)
+
+
+# Initialize the State object & Connect
+state = State(inventory, config)
+connect_all(state)
+
+# Add operations to state
+add_op(
+    state,
+    server.user,
+    user='vagrant',
+    home='/home/vagrant',
+    shell='/bin/bash',
+    sudo=True
+)
+
+# Run all state operations
+run_ops(state)
+
+# Get facts for all hosts
+facts = get_facts(state, LinuxName)
+```
+
+## Developing
```

Comment:
I think this should just link through to the contributing docs page like https://github.com/Fizzadar/pyinfra/blob/2.x/CONTRIBUTING.md - mainly to avoid having stuff docmented in more than one location.

But agree this should be in the readme by default!